src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -1815,7 +1816,7 @@ private String determineFilename(
         }
 
         final Date disbursementDate = extractTypeContext.getDisbursementDate();
-        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmssSSS", Locale.US);
```

Comment:
Adjusting this format to include milliseconds won't be enough on its own to guarantee filename uniqueness. Based on the `extractAchs()` and `extractChecks()` methods, the date that's used in the filename is being defined BEFORE iterating over the file creation loops, meaning this filename-building method will likely be reusing the same source date for a given set of Check/ACH files.

To get the filename uniqueness to work as expected, you may need to make one or more of the following changes:

* Redefine the source date for each iteration of the Check/ACH `for` loops. (Not sure whether this would interfere with the data being sent to the bank, though.)
* Modify the `determineFilename()` method so that it also examines the filesystem to see if the given file already exists. If it does, then make adjustments to the filename accordingly, such as by tweaking the millisecond portion or adding/adjusting a custom suffix.